src/app/ui/components/tabs/tabs.tsx

```diff
@@ -0,0 +1,66 @@
+import { forwardRef } from 'react';
+
+import * as RadixTabs from '@radix-ui/react-tabs';
+import { css } from 'leather-styles/css';
+
+const tabRootStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+const tabsListStyles = css({
+  flexShrink: 0,
+  display: 'flex',
+  position: 'relative',
+  _before: {
+    content: '""',
+    position: 'absolute',
+    bottom: 0,
+    height: '2px',
+    width: '100%',
+    bg: 'accent.border-default',
+    zIndex: 9,
+  },
+});
+
+const tabsTriggerStyles = css({
+  position: 'relative',
+  textStyle: 'label.01',
+  flex: 1,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  py: 'space.04',
+  userSelect: 'none',
+  color: 'accent.text-subdued',
+  _hover: { background: 'accent.component-background-hover' },
+  '&[data-state="active"]': {
+    color: 'accent.text-primary',
+    _before: {
+      content: '""',
+      position: 'absolute',
+      bottom: 0,
+      height: '2px',
+      width: '100%',
+      bg: 'accent.text-subdued',
+      zIndex: 10,
+    },
+  },
+  _focusVisible: { outline: 0, _before: { bg: 'lightModeBlue.500' } },
+});
+
+const Root: typeof RadixTabs.Root = forwardRef(props => (
```

Comment:
Is it nec to always forwardRef on these? I couldn't quite determine if always needed bc in their examples it wasn't clear to me. I was referencing this as an example: https://www.radix-ui.com/primitives/docs/components/dropdown-menu#custom-apis

Reply:
I am also unsure if we need to do it so it would be good to understand why we do if it is necessary

Reply:
UI libraries always export components with `forwardRef`, if there's an underlying native input element, as to allow the consumer access to that underlying element. See source of Radix/Chakra etc. Were this app code I'd say "only if we need it", as this is the base of our UI library that'll one day be in it's own package, I believe we should always do it (if the element has a ref).

Reply:
Why would you say, in the example I posted, they don't forwardRef to the `Root`, `Trigger`, `Label`, `Item`, `Group`, `RadioGroup`, or `Separator`. But, they do for `Content`, `CheckboxItem`, and `RadioItem`? Is that only bc they are showing an example that they anticipate those components will need access to the underlying element, while others may not?

Reply:
These are already `forwardRef`'d, and because they're not wrapping them (only renaming), it isn't necessary. 

```ts
// ✅ this is fine, the ref is retained
export const DropdownMenuGroup = DropdownMenuPrimitive.Group;
// ❌ ref lost
export function AltDropdownMenuGroup () {
  return <DropdownMenuPrimitive.Group {...} /}
}
```

Reply:
Got it, thx, I'll go back and check the two I made already and adjust. 👍

Reply:
Thanks Kyran 👏

Reply:
I wanted to double check this. Do I need to `forwardRef` for our [new Dialog](https://github.com/leather-wallet/extension/blob/e0ff6e67627d8de2b2006eb87273e692273a3962/src/app/ui/components/containers/dialog/dialog.tsx) (to replace `BaseDrawer` and built using `RadixDialog`.

That's just a dialog with no underlying inputs but it could receive slots / children that contain inputs / buttons etc.

Reply:
As a rule of thumb, if it has a ref, I think the safest option is to forward it. But in case of diaglog its less clear when we'd need it, versus inputs/buttons